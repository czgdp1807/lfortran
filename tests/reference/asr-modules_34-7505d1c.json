{
    "basename": "asr-modules_34-7505d1c",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_34.f90",
    "infile_hash": "345b669b534e06097d4a1202a471d3b9344402dc41e008388e6af962",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-modules_34-7505d1c.stdout",
    "stdout_hash": "b8ec623c1a9ceafbf27944ab09c67c8b4d87066c2a65a1cf76e131a7",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}