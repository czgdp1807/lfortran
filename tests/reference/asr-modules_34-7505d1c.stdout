(TranslationUnit
    (SymbolTable
        1
        {
            fpm_cmd_install_modules_34:
                (Module
                    (SymbolTable
                        2
                        {
                            build_model:
                                (ExternalSymbol
                                    2
                                    build_model
                                    4 build_model
                                    fpm_modules_34
                                    []
                                    build_model
                                    Public
                                )
                        })
                    fpm_cmd_install_modules_34
                    [fpm_modules_34]
                    .false.
                    .false.
                ),
            fpm_manifest_modules_34:
                (Module
                    (SymbolTable
                        8
                        {
                            package_config_t:
                                (ExternalSymbol
                                    8
                                    package_config_t
                                    10 package_config_t
                                    fpm_manifest_package_modules_34
                                    []
                                    package_config_t
                                    Public
                                )
                        })
                    fpm_manifest_modules_34
                    [fpm_manifest_package_modules_34]
                    .true.
                    .false.
                ),
            fpm_manifest_package_modules_34:
                (Module
                    (SymbolTable
                        10
                        {
                            package_config_t:
                                (StructType
                                    (SymbolTable
                                        11
                                        {
                                            name:
                                                (Variable
                                                    11
                                                    name
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Character 1 -2 () [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            version:
                                                (Variable
                                                    11
                                                    version
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        10 version_t
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    package_config_t
                                    [version_t]
                                    [name
                                    version]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                ),
                            version_t:
                                (ExternalSymbol
                                    10
                                    version_t
                                    13 version_t
                                    fpm_versioning_modules_34
                                    []
                                    version_t
                                    Public
                                )
                        })
                    fpm_manifest_package_modules_34
                    [fpm_versioning_modules_34]
                    .true.
                    .false.
                ),
            fpm_modules_34:
                (Module
                    (SymbolTable
                        4
                        {
                            1_package_config_t:
                                (ExternalSymbol
                                    4
                                    1_package_config_t
                                    10 package_config_t
                                    fpm_manifest_package_modules_34
                                    []
                                    package_config_t
                                    Public
                                ),
                            build_model:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            associate_block:
                                                (AssociateBlock
                                                    (SymbolTable
                                                        6
                                                        {
                                                            1_package_config_t_version:
                                                                (ExternalSymbol
                                                                    6
                                                                    1_package_config_t_version
                                                                    11 version
                                                                    1_package_config_t
                                                                    []
                                                                    version
                                                                    Public
                                                                ),
                                                            1_to_string:
                                                                (ExternalSymbol
                                                                    6
                                                                    1_to_string
                                                                    14 to_string
                                                                    1_version_t
                                                                    []
                                                                    to_string
                                                                    Public
                                                                ),
                                                            1_version_t:
                                                                (ExternalSymbol
                                                                    6
                                                                    1_version_t
                                                                    13 version_t
                                                                    fpm_versioning_modules_34
                                                                    []
                                                                    version_t
                                                                    Public
                                                                ),
                                                            fpm_versioning_modules_34_version_t:
                                                                (ExternalSymbol
                                                                    6
                                                                    fpm_versioning_modules_34_version_t
                                                                    13 version_t
                                                                    fpm_versioning_modules_34
                                                                    []
                                                                    version_t
                                                                    Public
                                                                ),
                                                            ii:
                                                                (Variable
                                                                    6
                                                                    ii
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Pointer
                                                                        (Integer 4 [])
                                                                    )
                                                                    Source
                                                                    Private
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    associate_block
                                                    [(=>
                                                        (Var 6 ii)
                                                        (Var 5 i)
                                                    )
                                                    (Print
                                                        ()
                                                        [(Var 6 ii)]
                                                        ()
                                                        ()
                                                    )
                                                    (SubroutineCall
                                                        6 1_to_string
                                                        ()
                                                        [((Var 5 version))]
                                                        (StructInstanceMember
                                                            (Var 5 package)
                                                            6 1_package_config_t_version
                                                            (Struct
                                                                6 fpm_versioning_modules_34_version_t
                                                                []
                                                            )
                                                            ()
                                                        )
                                                    )]
                                                ),
                                            i:
                                                (Variable
                                                    5
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            package:
                                                (Variable
                                                    5
                                                    package
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        4 package_config_t
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            version:
                                                (Variable
                                                    5
                                                    version
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Character 1 -2 () [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    build_model
                                    [1_to_string]
                                    [(Var 5 package)]
                                    [(AssociateBlockCall
                                        5 associate_block
                                    )
                                    (ImplicitDeallocate
                                        [5 version]
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            package_config_t:
                                (ExternalSymbol
                                    4
                                    package_config_t
                                    10 package_config_t
                                    fpm_manifest_package_modules_34
                                    []
                                    package_config_t
                                    Public
                                )
                        })
                    fpm_modules_34
                    [fpm_manifest_modules_34]
                    .true.
                    .false.
                ),
            fpm_versioning_modules_34:
                (Module
                    (SymbolTable
                        13
                        {
                            as_string:
                                (Function
                                    (SymbolTable
                                        15
                                        {
                                            1_to_string:
                                                (ExternalSymbol
                                                    15
                                                    1_to_string
                                                    14 to_string
                                                    version_t
                                                    []
                                                    to_string
                                                    Public
                                                ),
                                            self:
                                                (Variable
                                                    15
                                                    self
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Class
                                                        13 version_t
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            string:
                                                (Variable
                                                    15
                                                    string
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Character 1 -2 () [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    as_string
                                    [1_to_string]
                                    [(Var 15 self)]
                                    [(SubroutineCall
                                        15 1_to_string
                                        ()
                                        [((Var 15 string))]
                                        (Var 15 self)
                                    )]
                                    (Var 15 string)
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            char:
                                (GenericProcedure
                                    13
                                    char
                                    [13 as_string]
                                    Public
                                ),
                            to_string:
                                (Function
                                    (SymbolTable
                                        16
                                        {
                                            self:
                                                (Variable
                                                    16
                                                    self
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Class
                                                        13 version_t
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            string:
                                                (Variable
                                                    16
                                                    string
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Character 1 -2 () [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    to_string
                                    []
                                    [(Var 16 self)
                                    (Var 16 string)]
                                    []
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            version_t:
                                (StructType
                                    (SymbolTable
                                        14
                                        {
                                            num:
                                                (Variable
                                                    14
                                                    num
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Integer 4 [(()
                                                    ())])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            to_string:
                                                (ClassProcedure
                                                    14
                                                    to_string
                                                    ()
                                                    to_string
                                                    13 to_string
                                                    Source
                                                )
                                        })
                                    version_t
                                    []
                                    [num]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    fpm_versioning_modules_34
                    []
                    .true.
                    .false.
                ),
            modules_34:
                (Program
                    (SymbolTable
                        17
                        {
                            
                        })
                    modules_34
                    []
                    [(Print
                        ()
                        [(StringConstant
                            "running modules_34 program"
                            (Character 1 26 () [])
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
